bucketdelete/activity.go

```diff
@@ -0,0 +1,38 @@
+package bucketdelete
+
+import (
+	"context"
+	"fmt"
+
+	"go.artefactual.dev/tools/temporal"
+	"gocloud.dev/blob"
+	"gocloud.dev/gcerrors"
+)
+
+const Name = "bucket-delete"
+
+type (
+	Params struct {
+		Key string
+	}
+	Result   struct{}
+	Activity struct {
+		bucket *blob.Bucket
+	}
+)
+
+func New(bucket *blob.Bucket) *Activity {
+	return &Activity{bucket: bucket}
+}
+
+func (a *Activity) Execute(ctx context.Context, params *Params) (*Result, error) {
+	h := temporal.StartAutoHeartbeat(ctx)
```

Comment:
I had my conflicts about this, we recently saw in another project that may not always be true, and deletion of files from specific systems make take some time. I doubt it will be an issue in S3/MinIO buckets, but we may use this activity with other types of buckets. Also, the [`StartAutoHeartbeat`](https://github.com/artefactual-labs/gotools/blob/v0.14.0/temporal/heartbeat.go#L37-L40) won't do anything if you configure the activity option without a heartbeat timeout, I was planning to note that in the documentation ;)